python/caffe/_caffe.cpp

```diff
@@ -133,7 +133,7 @@ PySGDSolver::PySGDSolver(const string& param_file) {
   // as in PyNet, (as a convenience, not a guarantee), create a Python
   // exception if param_file can't be opened
   CheckFile(param_file);
-  solver_.reset(new SGDSolver<float>(param_file));
+  solver_ = boost::make_shared<SolvingDriver<float> >(param_file);
```

Comment:
@jeffdonahue 

See https://stackoverflow.com/questions/20895648/difference-in-make-shared-and-normal-shared-ptr-in-c